pkg/scheduler/objects/queue.go

```diff
@@ -1609,20 +1612,38 @@ func (sq *Queue) updateMaxResourceMetrics() {
 
 // updateAllocatedResourceMetrics updates allocated resource metrics for all queue types.
 func (sq *Queue) updateAllocatedResourceMetrics() {
+	// Special case when we prune the allocated resources to zero, we need to update the metrics
+	// here only keep basic types memory and vcores to make sure we have updated metrics
+	if sq.allocatedResource.IsEmpty() {
+		metrics.GetQueueMetrics(sq.QueuePath).SetQueueAllocatedResourceMetrics("memory", float64(0))
+		metrics.GetQueueMetrics(sq.QueuePath).SetQueueAllocatedResourceMetrics("vcores", float64(0))
+	}
```

Comment:
If we prune after we call the update for the metrics we keep the metrics clean and correct. We do not have to make a guess to what was zero. The queue also might not have used memory or vcores ever. Setting them here to zero in the metrics could throw things off.

This would need a comment in the code to why we first call the metric update and then prune.

Reply:
This is a better and clear solution @wilfred-s , addressed in latest PR.